runtime/compiler/env/VMJ9.cpp

```diff
@@ -6451,6 +6451,25 @@ TR_J9VMBase::getNumMethods(TR_OpaqueClassBlock * classPointer)
    return TR::Compiler->cls.romClassOf(classPointer)->romMethodCount;
    }
 
+/* This API takes a J9Class (TR_OpaqueClassBlock) and a J9Method (TR_OpaqueMethodBlock) and returns the
+ * index of the J9Method in the J9Class' array of J9Methods.
+ */
+uintptr_t
+TR_J9VMBase::getMethodIndexInClass(TR_OpaqueClassBlock *classPointer, TR_OpaqueMethodBlock *methodPointer)
+   {
+   void *methodsInClass = getMethods(classPointer);
+   uint32_t numMethods = getNumMethods(classPointer);
+
+   uintptr_t methodOffset = reinterpret_cast<uintptr_t>(methodPointer) - reinterpret_cast<uintptr_t>(methodsInClass);
+   TR_ASSERT_FATAL((methodOffset % sizeof (J9Method)) == 0, "methodOffset %p isn't a multiple of sizeof(J9Method)\n", methodOffset);
+
+   uintptr_t methodIndex = methodOffset / sizeof (J9Method);
+   TR_ASSERT_FATAL(methodIndex < numMethods, "methodIndex %p greater than numMethods %d for method %p in class %p\n",
```

Comment:
Should this format string (and the one just above) be using `%p` for integers?

I'd like to recommend `PRIxPTR`, but the format string is used for both `vfprintf()` and `omrstr_vprintf()`, which don't agree on the interpretation of all format specifiers. In particular, `omrstr_vprintf()` reads a 32-bit integer for `%lx` even on LP64 systems, where `PRIxPTR` could be `"lx"`.

I think the safest way to format a pointer-sized integer here is to cast it to `long long` (which should be 64-bit on all of our platforms) and use `%llx`.

Reply:
I"ll cast it to `uint64_t` (just in case) and use `%llx` as you suggest.